src/main/haskell/kore/test/Test/Kore/Builtin/Set.hs

```diff
@@ -264,13 +272,110 @@ prop_unifyFramingVariable set n =
+    concrete = asPattern $ Set.fromList [key]
+    symbolic = asSymbolicPattern $ Set.fromList [mkVar x]
+    original =
+        mkAnd
+            (mkPair Test.Int.intSort setSort (Test.Int.asPattern 1) concrete)
+            (mkPair Test.Int.intSort setSort (mkVar x) symbolic)
+    expected =
+        Predicated
+            { term =
+                mkPair Test.Int.intSort setSort
+                    symbolicKey
+                    (asSymbolicPattern $ Set.fromList [symbolicKey])
+            , predicate = Predicate.makeTruePredicate
+            , substitution =
+                [ (x, symbolicKey) ]
+            }
+    actual = evaluate original
+
+{- | Unify a concrete Set with symbolic-keyed Set in an axiom
+
+Apply the axiom
+@
+(x, [x]) => x
+@
+to the configuration
+@
+(1, [1])
+@
+yielding @1@.
+
+Iterated unification must turn the symbolic key @x@ into a concrete key by
+unifying the first element of the pair. This also requires that Set unification
+return a partial result for unifying the second element of the pair.
+
+ -}
+unit_concretizeKeysAxiom :: Assertion
+unit_concretizeKeysAxiom =
+    assertEqual "" expected actual
+  where
+    x = mkIntVar (testId "x")
+    key = 1
+    symbolicKey = Test.Int.asPattern key
+    symbolicSet = asSymbolicPattern $ Set.fromList [x]
+    concreteSet = asPattern $ Set.fromList [key]
+    axiom =
+        AxiomPattern
+            { axiomPatternLeft = mkPair Test.Int.intSort setSort x symbolicSet
+            , axiomPatternRight = x
+            , axiomPatternRequires = Predicate.makeTruePredicate
+            , axiomPatternAttributes = Default.def
+            }
+    config = evaluate $ mkPair Test.Int.intSort setSort symbolicKey concreteSet
+    expected =
+        Right
+            ( Predicated
+                { term = symbolicKey
+                , predicate =
+                    -- The predicate is not discharged because we do not
+                    -- provide functionality axioms for elementMap.
+                    give testSymbolOrAliasSorts
+                    Predicate.makeCeilPredicate
+                    $ asSymbolicPattern
+                    $ Set.fromList [symbolicKey]
+                , substitution = []
+                }
+            , mconcat
+                [ stepProof (StepProofVariableRenamings [])
+                , stepProof (StepProofUnification EmptyUnificationProof)
+                ]
+            )
+    actual = runStep config axiom
+
 -- | Specialize 'Set.asPattern' to the builtin sort 'setSort'.
 asPattern :: Set Integer -> CommonPurePattern Object
 Right asPattern = (. Set.map Test.Int.asConcretePattern) <$> Set.asPattern indexedModule setSort
 
 -- | Specialize 'Set.asPattern' to the builtin sort 'setSort'.
-asExpandedPattern :: Set.Builtin -> CommonExpandedPattern Object
-Right asExpandedPattern = Set.asExpandedPattern indexedModule setSort
+asExpandedPattern :: Set Integer -> CommonExpandedPattern Object
+Right asExpandedPattern =
+    (<$>)
```

Comment:
I'd prefer `fmap` here as mentioned above. I'm not a huge fan of excessive pointfree usage, and this is close to the limit. I'd probably not use pointfree here, but I don't mind that you did. Do we have some sort of clear style guide about this? @virgil-serbanuta ?

Reply:
`<$>` is a synonym for `fmap`, they are both pointfree. FWIW I agree with tom that the name `fmap` is ugly and generally prefer `<$>`. Maybe because operators are visually distinct from alphanumeric functions, and it helps signal that there's a functor involved.